File: CS/DXSample.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Reports;
using DevExpress.ExpressApp;

namespace DXSample.Module {
    public class Updater : ModuleUpdater {
        public Updater(ObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            CreateReport("MasterReport");
            CreateReport("DetailReport");
        }
        private void CreateReport(string reportName) {
            ReportData reportdata = ObjectSpace.FindObject<ReportData>(new BinaryOperator("Name", reportName));
            if (reportdata == null) {
                reportdata = ObjectSpace.CreateObject<ReportData>();
                XafReport rep = new XafReport();
                rep.ReportName = reportName;
                rep.ObjectSpace = ObjectSpace;
                rep.LoadLayout(GetType().Assembly.GetManifestResourceStream(
                   "DXSample.Module.EmbeddedReports." + reportName + ".repx"));
                reportdata.SaveXtraReport(rep);
                reportdata.Save();
            }
        }
    }
}

File: VB/DXSample.Module/Updater.vb
Imports System
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.ExpressApp

Namespace DXSample.Module

    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub

        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            CreateReport("MasterReport")
            CreateReport("DetailReport")
        End Sub

        Private Sub CreateReport(ByVal reportName As String)
            Dim reportdata As ReportData = ObjectSpace.FindObject(Of ReportData)(New BinaryOperator("Name", reportName))
            If reportdata Is Nothing Then
                reportdata = ObjectSpace.CreateObject(Of ReportData)()
                Dim rep As XafReport = New XafReport()
                rep.ReportName = reportName
                rep.ObjectSpace = ObjectSpace
                rep.LoadLayout([GetType]().Assembly.GetManifestResourceStream("DXSample.Module.EmbeddedReports." & reportName & ".repx"))
                reportdata.SaveReport(rep)
                reportdata.Save()
            End If
        End Sub
    End Class
End Namespace
